src/android.c

```diff
@@ -0,0 +1,402 @@
+        &osa->outputMixObject, 0, 0, 0))
+    {
+        return SoundIoErrorOpeningDevice;
+    }
+
+    if(SL_RESULT_SUCCESS != (*osa->outputMixObject)->Realize(osa->outputMixObject,
+        SL_BOOLEAN_FALSE))
+    {
+        outstream_destroy_android(si, os);
+        return SoundIoErrorOpeningDevice;
+    }
+
+    SLDataLocator_AndroidSimpleBufferQueue loc_bufq = {
+        SL_DATALOCATOR_ANDROIDSIMPLEBUFFERQUEUE,
+        // TODO: The minimum buffer count is 1. When would you want to have
+        // a single buffer?
+        2
+    };
+
+    SLAndroidDataFormat_PCM_EX format_pcm = {
+        SL_ANDROID_DATAFORMAT_PCM_EX,
+        outstream->layout.channel_count,
+        outstream->sample_rate * 1000,
+        outstream->bytes_per_sample * 8,
+        outstream->bytes_per_sample * 8,
+        outstream->layout.channel_count == 1 ? SL_SPEAKER_FRONT_CENTER :
+            (SL_SPEAKER_FRONT_LEFT | SL_SPEAKER_FRONT_RIGHT),
+        SL_BYTEORDER_LITTLEENDIAN,
+        outstream->format == SoundIoFormatFloat32LE ?
+            SL_ANDROID_PCM_REPRESENTATION_FLOAT :
+            outstream->format == SoundIoFormatS16LE ?
+            SL_ANDROID_PCM_REPRESENTATION_SIGNED_INT :
+            SL_ANDROID_PCM_REPRESENTATION_UNSIGNED_INT
+    };
+
+    SLDataSource audioSrc = {&loc_bufq, &format_pcm};
+    SLDataLocator_OutputMix loc_outmix = {
+        SL_DATALOCATOR_OUTPUTMIX,
+        osa->outputMixObject
+    };
+    SLDataSink audioSnk = {&loc_outmix, NULL};
+    SLInterfaceID ids[1] = {SL_IID_BUFFERQUEUE};
+    const SLboolean req[1] = {SL_BOOLEAN_TRUE};
+
+    if (SL_RESULT_SUCCESS != (*sia->engineEngine)->CreateAudioPlayer(sia->engineEngine,
+        &osa->playerObject, &audioSrc, &audioSnk, 1, ids, req))
+    {
+        outstream_destroy_android(si, os);
+        return SoundIoErrorOpeningDevice;
+    }
+    if (SL_RESULT_SUCCESS != (*osa->playerObject)->Realize(osa->playerObject,
+        SL_BOOLEAN_FALSE))
+    {
+        outstream_destroy_android(si, os);
+        return SoundIoErrorOpeningDevice;
+    }
+
+    if (SL_RESULT_SUCCESS != (*osa->playerObject)->GetInterface(osa->playerObject,
+        SL_IID_PLAY, &osa->playerPlay))
+    {
+        outstream_destroy_android(si, os);
+        return SoundIoErrorOpeningDevice;
+    }
+
+    if (SL_RESULT_SUCCESS != (*osa->playerObject)->GetInterface(osa->playerObject,
+        SL_IID_BUFFERQUEUE, &osa->playerBufferQueue))
+    {
+        outstream_destroy_android(si, os);
+        return SoundIoErrorOpeningDevice;
+    }
+    if (SL_RESULT_SUCCESS != (*osa->playerBufferQueue)->RegisterCallback(
+        osa->playerBufferQueue, write_callback_android, os))
+    {
+        outstream_destroy_android(si, os);
+        return SoundIoErrorOpeningDevice;
+    }
+
+    // TODO: An audio stream's latency and jitter can be reduced by using the
+    // native buffer size. Unfortunately, there is no officially supported way
+    // to get the native buffer size without access to a JVM.
```

Comment:
I've been getting around this by requesting a latency of `FACTOR OF BUFFER SIZE / SAMPLING RATE`, which seems to work.

It would've been nice to set the default latency to something that results in the native buffer size, though.